gapir/cc/memory_allocator_test.cpp

```diff
@@ -0,0 +1,295 @@
+#include <gtest/gtest.h>
+
+#include "memory_allocator.h"
+
+using namespace gapir;
+
+void TEST_ALLOCATIONS(const std::vector<MemoryAllocator::Handle>& allocations,
+                      const std::vector<size_t>& allocationSizes,
+                      const std::vector<bool>& expectPass) {
+  EXPECT_EQ(allocations.size(), allocationSizes.size());
+  EXPECT_EQ(allocations.size(), expectPass.size());
+
+  for (unsigned int i = 0; i < allocations.size(); ++i) {
+    EXPECT_EQ(allocations[i] != nullptr, expectPass[i]);
+
+    for (unsigned int j = 0; j < allocations.size(); ++j) {
+      if (allocations[j] != nullptr) {
+        for (unsigned int k = 0; k < allocationSizes[j]; ++k) {
+          allocations[j][k] = 0;
+        }
+      }
+    }
+
+    if (allocations[i] != nullptr) {
+      for (unsigned int k = 0; k < allocationSizes[i]; ++k) {
+        allocations[i][k] = 255;
+      }
+    }
+
+    bool passedContaminationCheck = true;
+
+    for (unsigned int j = 0; j < allocations.size(); ++j) {
+      if (allocations[j] != nullptr) {
+        for (unsigned int k = 0; k < allocationSizes[j]; ++k) {
+          passedContaminationCheck &=
+              (int)(j == i ? 255 : 0) == (int)allocations[j][k];
+        }
+      }
+    }
+
+    EXPECT_TRUE(passedContaminationCheck);
+  }
+}
+
+TEST(MemoryAllocator, SimpleStaticAllocate) {
+  std::unique_ptr<MemoryAllocator> allocator = MemoryAllocator::create(2048);
+  EXPECT_TRUE(allocator->allocateStatic(1024) != nullptr);
+}
+
+TEST(MemoryAllocator, StaticAllocateTooMuch) {
+  std::unique_ptr<MemoryAllocator> allocator = MemoryAllocator::create(2048);
+  EXPECT_TRUE(allocator->allocateStatic(8192) == nullptr);
+}
+
+TEST(MemoryAllocator, SimpleMultipleStaticAllocate) {
+  std::vector<size_t> allocationSizes = {1024, 512, 256, 128, 64, 32, 16,
+                                         8,    4,   2,   1,   1,  1};
+  std::vector<bool> expectPass = {true, true, true, true, true, true, true,
+                                  true, true, true, true, true, false};
+
+  std::vector<MemoryAllocator::Handle> addresses;
+
+  std::unique_ptr<MemoryAllocator> allocator = MemoryAllocator::create(2048);
+
+  for (unsigned int i = 0; i < allocationSizes.size(); ++i) {
+    addresses.push_back(allocator->allocateStatic(allocationSizes[i]));
+  }
+
+  TEST_ALLOCATIONS(addresses, allocationSizes, expectPass);
+}
+
+TEST(MemoryAllocator, SimpleGrowStaticAllocate) {
+  std::unique_ptr<MemoryAllocator> allocator = MemoryAllocator::create(2048);
```

Comment:
You seem to use `2048` across all tests, I reckon it would make sense to add `#define MEMORY_SIZE 2048` and use this macro constant to replace all hardcoded 2048 ?